influunt-api/modules/influunt-core/app/engine/GerenciadorDeEstagiosHelper.java

```diff
@@ -54,4 +55,30 @@ public static void terminaTempoEstagio(RangeMap<Long, IntervaloEstagio> interval
             intervalos.put(novoRange, intervalo);
         }
     }
+
+    public static void aumentarTempoEstagio(RangeMap<Long, IntervaloEstagio> intervalos,
+                                                 long contadorIntervalo, long tempo) {
+        Map.Entry<Range<Long>, IntervaloEstagio> range = intervalos.getEntry(contadorIntervalo - 100L);
+        final IntervaloEstagio intervalo = range.getValue();
+
+        if (intervalo.getDuracao() < tempo) {
+            long novaDuracao = tempo;
+
+            intervalos.remove(range.getKey());
+            intervalo.setDuracao(novaDuracao);
+            intervalos.put(Range.closedOpen(range.getKey().lowerEndpoint(), range.getKey().lowerEndpoint() + novaDuracao),
+                intervalo);
+        }
+    }
+
+    public static boolean isCumpriTempoVerdeSegurancao(List<EstagioPlano> lista) {
```

Comment:
acho que o nome seria `deveCumprirTempoVerdeSeguranca` ou `isCumpreVerdeSeguranca` (não sei quanto ao `is`).
Está errado aqui o "cumpri" e o "segurancao"